prov/util/src/util_shm.c

```diff
@@ -68,21 +68,31 @@ static void smr_peer_addr_init(struct smr_addr *peer)
 void smr_cma_check(struct smr_region *smr, struct smr_region *peer_smr)
 {
 	struct iovec local_iov, remote_iov;
+	int remote_pid;
 	int ret;
 
-	if (peer_smr->cma_cap != SMR_CMA_CAP_NA) {
-		smr->cma_cap = peer_smr->cma_cap;
+	if (smr != peer_smr && !(peer_smr->cma_cap & SMR_PEER_CMA_CAP_NA)) {
+		smr->cma_cap = (peer_smr->cma_cap & ~0xf) + (smr->cma_cap & 0xf);
 		return;
 	}
-	local_iov.iov_base = &smr->cma_cap;
-	local_iov.iov_len = sizeof(smr->cma_cap);
+
+	remote_pid = peer_smr->pid;
+	local_iov.iov_base = &remote_pid;
+	local_iov.iov_len = sizeof(remote_pid);
 	remote_iov.iov_base = (char *)peer_smr->base_addr +
-			      ((char *)&peer_smr->cma_cap - (char *)peer_smr);
-	remote_iov.iov_len = sizeof(peer_smr->cma_cap);
+			      ((char *)&peer_smr->pid - (char *)peer_smr);
+	remote_iov.iov_len = sizeof(peer_smr->pid);
 	ret = ofi_process_vm_writev(peer_smr->pid, &local_iov, 1,
 				    &remote_iov, 1, 0);
-	smr->cma_cap = (ret == -1) ? SMR_CMA_CAP_OFF : SMR_CMA_CAP_ON;
-	peer_smr->cma_cap = smr->cma_cap;
+	assert(remote_pid == peer_smr->pid);
+	if (smr == peer_smr) {
+		smr->cma_cap &= ~SMR_SELF_CMA_CAP_NA;
+		smr->cma_cap |= (ret == -1) ? SMR_SELF_CMA_CAP_OFF : SMR_SELF_CMA_CAP_ON;
+	} else {
+		smr->cma_cap &= ~SMR_PEER_CMA_CAP_NA;
+		smr->cma_cap |= (ret == -1) ? SMR_PEER_CMA_CAP_OFF : SMR_PEER_CMA_CAP_ON;
+		peer_smr->cma_cap = (smr->cma_cap & ~0xf) + (peer_smr->cma_cap & 0xf);
```

Comment:
Similarly I think this can be simplified to
`peer_smr->cma_cap |= (ret == -1) ? SMR_PEER_CMA_CAP_OFF : SMR_PEER_CMA_CAP_ON`